### DISCUSSION TITLE
after upgrading to latest 22 from 19.3 -> schema validation works differntly and fails my queries/mutations

### DISCUSSION THREAD

**Discussion by guykatz** [graphql-java/graphql-java#3780] [Q&A] [answered]
Hi all'
historicly i have a situation where the client side mistaikingly sometimes sends strings where the gql schema has Int type.
in 19.3 it worked fine and we assumed there is an automatic conversion between the incoming string to the expected Int ni the gql-java engine.
after upgrading to 22 this now started to break and all of these instances return back this error:
`Expected a value that can be converted to type 'Int' but it was a 'String'`
is this the expected behavior in 22?
and most importantly, can I change some flag to return to the old way of automatic converting?
thanks!
attached in the GQL mutation with the field 'applicationID' that now fails... (we have many of these)
![image](https://github.com/user-attachments/assets/5d21007d-60d0-4d7d-9e4e-946e327ce716)

**Comment by guykatz** [comment]
found answer in the release notes. sorry and 10x
more info here: [https://github.com/graphql-java/graphql-java/pull/3553](https://github.com/graphql-java/graphql-java/pull/3553)

**Reply by dondonz** [reply]
Hello, thanks for opening the discussion, and I'm glad you found the answer. However that is a signal to me that I could make this clearer in the documentation. Thanks!

**Reply by guykatz** [reply]
thanks @dondonz - BTW, i couldnt find the proper way to "use" the new legacyCoercingInputInterceptor class or its InputInterceptor ifc...
is this somewhere in the docs?
10x again

**Reply by dondonz** [reply, marked as answer]
No it isn't in the docs, which also should be fixed!

I'll also write a code snippet with this

**Reply by dondonz** [reply]
Hello again! Here's a preview of the documentation I've drafted: https://github.com/graphql-java/graphql-java-page/pull/186/files

Let me know what you think.
